tests/DidCache.spec.ts

```diff
@@ -0,0 +1,126 @@
+import * as Base58 from 'bs58';
+import { Cas } from '../src/Cas';
+import { DidCache } from '../src/DidCache';
+import { DidDocument } from '@decentralized-identity/did-common-typescript';
+import { WriteOperation } from '../src/Operation'
+
+// Implementation of Did document update - no-op for testing purposes
+function didDocumentUpdate (didDoc: DidDocument, _operation: WriteOperation): DidDocument {
+  return didDoc;
+}
+
+const didDocJson = {
+  "@context": "https://w3id.org/did/v1",
+  "id": "did:sidetree:ignored",
+  "publicKey": [{
+    "id": "did:sidetree:didPortionIgnored#key-1",
+    "type": "RsaVerificationKey2018",
+    "owner": "did:sidetree:ignoredUnlessResolvable",
+    "publicKeyPem": "-----BEGIN PUBLIC KEY...END PUBLIC KEY-----\r\n"
+  }],
+  "service": [{
+    "type": "IdentityHub",
+    "publicKey": "did:sidetree:ignored#key-1",
+    "serviceEndpoint": {
+      "@context": "schema.identity.foundation/hub",
+      "@type": "UserServiceEndpoint",
+      "instances": ["did:bar:456", "did:zaz:789"]
+    }
+  }]
+};
+
+// Implementation of Did document create - return a dummy did document
+function didDocumentCreate (_operation: WriteOperation): DidDocument {
+  return new DidDocument(didDocJson);
+}
+
+
+// Implementation of a dummy cas class for testing - a simple hash map
+class DummyCas implements Cas {
+  bufs: Buffer[] = [];
+
+  public async write (content: Buffer): Promise<string> {
+    this.bufs.push(content);
+    return (this.bufs.length - 1).toString();
+  }
+
+  public async read (address: string): Promise<Buffer> {
+    const idx = +address;
```

Comment:
What does idx stand for?

Reply:
index (into the bufs array) - added a comment to this effect.